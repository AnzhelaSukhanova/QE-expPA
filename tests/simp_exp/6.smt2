(set-option :produce-models true)
(set-logic BV)
(declare-fun y () (_ BitVec 4))
(declare-fun z () (_ BitVec 4))
(declare-fun t () (_ BitVec 4))
(assert
 (exists ((x (_ BitVec 4))) (and 
 (or (bvult (bvshl #b0001 x) (bvmul #b0111 y)) (= (bvshl #b0001 x) (bvmul #b0111 y))) 
 (or (bvult (bvshl #b0001 x) z) (= (bvshl #b0001 x) z)) 
 (or (bvult (bvshl #b0001 x) (bvshl #b0001 t)) (= (bvshl #b0001 x) (bvshl #b0001 t))))))
