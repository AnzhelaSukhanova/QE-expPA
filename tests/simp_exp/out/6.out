(let (($e1 (bvmul y #x7))) (let (($e2 (bvshl #x1 t))) (let (($e3 (= $e2 #x0))) (let (($e4 (ite (bvult $e2 #x8) (ite (bvult $e2 #x4) (ite (bvult $e2 #x2) (ite $e3 #xf #x0) #x1) #x2) #x3))) (let (($e5 (bvshl #x1 $e4))) (let (($e6 (bvshl #x1 (bvadd #x1 $e4)))) (let (($e7 (= #x1 $e2))) (let (($e8 (= $e2 #x2))) (let (($e9 (= $e2 #x4))) (let (($e10 (= #x8 $e2))) (let (($e11 (bvult #x1 $e2))) (let (($e12 (bvult #x2 $e2))) (let (($e13 (bvult #x4 $e2))) (let (($e14 (bvult #x8 $e2))) (let (($e15 (= $e1 #x0))) (let (($e16 (ite (bvult $e1 #x8) (ite (bvult $e1 #x4) (ite (bvult $e1 #x2) (ite $e15 #xf #x0) #x1) #x2) #x3))) (let (($e17 (bvshl #x1 $e16))) (let (($e18 (bvshl #x1 (bvadd #x1 $e16)))) (let (($e19 (= #x1 $e1))) (let (($e20 (= $e1 #x2))) (let (($e21 (= $e1 #x4))) (let (($e22 (= #x8 $e1))) (let (($e23 (bvult #x1 $e1))) (let (($e24 (bvult #x2 $e1))) (let (($e25 (bvult #x4 $e1))) (let (($e26 (bvult #x8 $e1))) (let (($e27 (= z #x0))) (let (($e28 (ite (bvult z #x8) (ite (bvult z #x4) (ite (bvult z #x2) (ite $e27 #xf #x0) #x1) #x2) #x3))) (let (($e29 (= z #x1))) (let (($e30 (= z #x2))) (let (($e31 (= z #x4))) (let (($e32 (= z #x8))) (let (($e33 (bvult #x1 z))) (let (($e34 (bvult #x2 z))) (let (($e35 (bvult #x4 z))) (let (($e36 (bvult #x8 z))) (and (not (and (not $e24) (not $e23) (not $e20) (not $e19) (not $e25) (not $e21) (not $e26) (not $e22) $e15 (not $e15) (not $e23) (not $e19) (not $e24) (not $e20) (not $e25) (not $e21) (not $e26) (not $e22) $e15 (not $e15) (not $e23) (not $e19) (not (bvult $e18 $e1)) (not (bvult $e17 $e1)) (not $e24) (not (= $e1 $e18)) (not (= $e1 $e17)) (not $e20))) (not (and (not $e12) (not $e11) (not $e8) (not $e7) (not $e13) (not $e9) (not $e14) (not $e10) $e3 (not $e3) (not $e11) (not $e7) (not $e12) (not $e8) (not $e13) (not $e9) (not $e14) (not $e10) $e3 (not $e3) (not $e11) (not $e7) (not (bvult $e6 $e2)) (not (bvult $e5 $e2)) (not $e12) (not (= $e2 $e6)) (not (= $e2 $e5)) (not $e8))) (not (and (not $e34) (not $e33) (not $e35) (not $e30) (not $e29) (not $e36) (not $e31) $e27 (not $e32) (not $e33) (not $e27) (not $e34) (not $e29) (not $e35) (not $e30) (not $e36) (not $e31) $e27 (not $e32) (not $e33) (not $e27) (not (bvult (bvshl #x1 (bvadd #x1 $e28)) z)) (not (bvult (bvshl #x1 $e28) z)) (not $e34) (not $e29) (not $e30))))))))))))))))))))))))))))))))))))))))
